src/providers/CookieConsent.tsx

```diff
@@ -0,0 +1,152 @@
+import { UseGdprMetadata, useGdprMetadata } from '../hooks/useGdprMetadata'
+import { useSiteMetadata } from '../hooks/useSiteMetadata'
+
+export enum CookieConsentStatus {
+  NOT_AVAILABLE = -1,
+  APPROVED = 0,
+  REJECTED = 1
+}
+
+export interface ConsentStatus {
+  [name: string]: CookieConsentStatus
+}
+
+interface ConsentProviderValue {
+  cookies: UseGdprMetadata['optionalCookies']
+  cookieConsentStatus: ConsentStatus
+  setConsentStatus: (cookieName: string, status: CookieConsentStatus) => void
+  resetConsentStatus: (status?: CookieConsentStatus) => void
+}
+
+const ConsentContext = createContext({} as ConsentProviderValue)
+
+function ConsentProvider({ children }: { children: ReactNode }): ReactElement {
+  const cookies = useGdprMetadata()
+
+  const { privacyPreferenceCenter } = useSiteMetadata().appConfig
+
+  const [consentStatus, setConsentStatus] = useState({} as ConsentStatus)
+
+  function resetConsentStatus(status = CookieConsentStatus.NOT_AVAILABLE) {
+    const resetCookieConsent = {} as ConsentStatus
+    cookies.optionalCookies?.map((cookie) => {
+      deleteCookie(cookie.cookieName)
+      resetCookieConsent[cookie.cookieName] = status
+    })
+    setConsentStatus(resetCookieConsent)
+  }
+
+  function setCookieConsentStatus(
+    cookieName: string,
+    status: CookieConsentStatus
+  ) {
+    setConsentStatus({ ...consentStatus, [cookieName]: status })
+  }
+
+  function handleAccept(cookieName: string) {
+    setCookie(cookieName, true)
+    switch (cookieName) {
+      case 'AnalyticsCookieConsent':
+        break
+      default:
+        break
+      // Add your specific logic here
+      // e.g.
+      /* function handleAnalytics() {
+          ReactGA.initialize(analyticsId)
+          ReactGA.pageview(window.location.pathname + window.location.search)
+        } */
+    }
+  }
+
+  function handleReject(cookieName: string) {
+    setCookie(cookieName, false)
+    switch (cookieName) {
+      case 'AnalyticsCookieConsent':
+        break
+      default:
+        break
+      // Add your specific logic here
+    }
+  }
+
+  useEffect(() => {
+    if (!privacyPreferenceCenter) return
+
+    const initialValues = {} as ConsentStatus
+    cookies.optionalCookies?.map((cookie) => {
+      const cookieValue = getCookieValue(cookie.cookieName)
+
+      switch (cookieValue) {
+        case 'true':
+          initialValues[cookie.cookieName] = CookieConsentStatus.APPROVED
+          break
+        case 'false':
+          initialValues[cookie.cookieName] = CookieConsentStatus.REJECTED
+          break
+        default:
+          initialValues[cookie.cookieName] = CookieConsentStatus.NOT_AVAILABLE
+          break
+      }
+    })
+
+    setConsentStatus(initialValues)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (!privacyPreferenceCenter || !window || !navigator) return
+
+    if (
+      (window.doNotTrack && window.doNotTrack === '1') || // Safari >7.1.3, Edge
```

Comment:
Do Not Track never worked as expected and the standard has been abandoned, see https://en.wikipedia.org/wiki/Do_Not_Track

So we don't need to bother with detecting it

Reply:
It actually works in edge, really good i might say.

Reply:
I think this was the beginning of the end of that standard when Microsoft had decided in Edge to turn it on by default so immediately it was not reflecting user intention anymore, but browser vendor decisions, making it kinda useless. Most browsers by now have switched to some built-in tracking blocker anyway

Reply:
I just committed some changes including the removal of Do Not Track detection